lib/tpm2_util.c

```diff
@@ -132,42 +132,71 @@ int tpm2_util_hex_to_byte_structure(const char *inStr, UINT16 *byteLength,
     return 0;
 }
 
-void tpm2_util_hexdump(BYTE *data, size_t len, bool plain) {
+void tpm2_util_hexdump(const BYTE *data, size_t len, bool plain) {
+    FILE *fd = fmemopen((void*)data, len, "r");
+    if (!fd) {
+        /* TODO: should we change the API to allow error? */
```

Comment:
AFAICT, the only caller of this function currently is tpm2_util_public_to_yaml(), so it should be quite easy to change the function signature to propagate the error.

That said, I wonder if instead of using fmemopen() and calling tpm2_util_hexdump_file(), can we do the opposite and make tpm2_util_hexdump_file() read the file contents and call tpm2_util_hexdump()?

Since once the data is in memory, tpm2_util_hexdump() shouldn't really have an error path.

Reply:
I think I prefer it this way, as potentially big buffers won't need to be fully read into memory. Granted, most tpm data structures are quite small. Either way is fine by me in the end.